### PR TITLE
Improve check invoice performance

### PR DESCRIPTION
**Pull request by tonyaraujop**
closes: https://github.com/comarev/comarev/issues/66

#### What?

- Added a conditional block to check invoices paid

#### Why?

- To #66 

#### How to test it?

- Make a GET request to find companies with invoices paid

### CODE REVIEW

File: app/controllers/invoices_controller.rb [hunk 1]
```diff
@@ -27,9 +27,12 @@ def update
...
   def check
     authorize(Invoice)
-    @company = Company.find_by!(code: params[:code])
 
-    return head :unprocessable_entity unless policy_scope(Invoice).all?(&:paid?)
+    if policy_scope(Invoice).all?(&:paid?)
+      @company = Company.find_by!(code: params[:code])
+    else
+      return head :unprocessable_entity
+    end
```
Comment (m-pereira) [inline app/controllers/invoices_controller.rb:35]:
```suggestion
    return head :unprocessable_entity if policy_scope(Invoice).all?(&:paid?) 
    
    @company = Company.find_by!(code: params[:code])
```

**Review by m-pereira — COMMENTED** [review]
Hey, @tonyaraujop thanks for this PR, you got the idea!

File: app/controllers/invoices_controller.rb (hunk 1 above)
Comment (tonyaraujop) [reply app/controllers/invoices_controller.rb]: Hey @m-pereira I make the suggested changes and also double check if all the tests were covered, below you can see a screen shot of it.

image

Comment (m-pereira) [reply app/controllers/invoices_controller.rb]: Actually, we're using [simplecov](https://github.com/simplecov-ruby/simplecov) to ensure the coverage. Can you please check if there is a test for this action

Comment (tonyaraujop) [reply app/controllers/invoices_controller.rb]: There are tests for this action (Check) and simplecov shows that 98.21% are covered.

**Review by m-pereira — APPROVED** [review]

**Comment by m-pereira** [comment]
@allcontributors add @tonyaraujop for code

**Pull request merged**